core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendUtils.scala

```diff
@@ -21,13 +21,27 @@ import org.apache.spark.SparkEnv
 import org.apache.spark.h2o.H2OConf
 import org.apache.spark.h2o.backends.{ArgumentBuilder, SharedBackendUtils}
 import org.apache.spark.h2o.utils.NodeDesc
-import water.{ExternalFrameUtils, H2O}
+import water.{ExternalFrameUtils, H2O, Paxos}
 
 /**
   * Various helper methods used in the external backend
   */
 private[external] trait ExternalBackendUtils extends SharedBackendUtils {
 
+  protected def waitForCloudSize(x: Int, ms: Long): Int = {
+    val start = System.currentTimeMillis()
+    while (System.currentTimeMillis() - start < ms) {
+      if (H2O.CLOUD.size() >= x && Paxos._commonKnowledge) {
```

Comment:
should it be `H2O.CLOUD.size() = x` ? or H2O.CLOUD.size() > expected_size is an okay state too?

Reply:
just for understanding - what is purpose of `Paxos._commonKnowledge` ?
thanks!

Reply:
If the cloud is bigger than expected size, it indicates that client is able connect to H2O nodes. IMHO, I don't think it's a problem.

From `Paxos` class:
```
 Paxos detects and builds a "cloud" - a cooperating group of nodes, with mutual
 knowledge of each other.  Basically tracks all the nodes that *this* node
 has ever heard of, and when *all* of the other nodes have all heard of each
 other, declares the situation as "commonKnowledge", and a Cloud.
```